### PR TITLE
Fix PR comments UI

### LINKED ISSUE #17768: Review comment cannot be edited
### Gitea Version

1.16.0+dev-581-gbaed01f24

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

try.gitea.io

### Database

_No response_

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

It does not currently seem possible to edit the main comment you can add when submitting a PR review.
Such a comment also does not have the labels (like Poster / Owner), reactions and other actions.

https://try.gitea.io/jpraet-org/test-repo/pulls/2

### Screenshots

ok (a regular comment on a PR):
image

not ok (a comment from submitted review cannot be edited):
image

not ok (a comment from submitted review cannot be edited, whereas the individual code comment can be edited):
image

### LINKED ISSUE #18315: Changing PR Comment Resolved State Disables Further Changes
### Gitea Version

1.16.0+dev-863-g076cead40

### Operating System

Kubuntu 20.04

### Browser Version

Firefox 96.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Description

When I change a PR comment in terms of it being `resolved`, I cannot (un-)resolve it any further without reloading the page.
What I mean by that:
Resolve a comment, and then try to unresolve it. It won't work.
The same is true when unresolving a resolved comment and then trying to re-resolve it.

The corresponding button seems to now be nothing more than a glorified label, especially regarding that the button text will be selected when pressing it.

### Screenshots

Not possible as simply _nothing_ happens.

### PR DESCRIPTION
**Pull request by wxiaoguang**
Closes: 
* https://github.com/go-gitea/gitea/issues/17768
* https://github.com/go-gitea/gitea/issues/18315

image

ps: this PR is still not ideal, some code are just copied & pasted, and there are still bugs with reactions (very old bugs). For 1.16-rc, this PR should be enough.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: routers/web/repo/issue.go [hunk 1]
```diff
@@ -2194,7 +2194,8 @@ func UpdateCommentContent(ctx *context.Context) {
 	if !ctx.IsSigned || (ctx.User.ID != comment.PosterID && !ctx.Repo.CanWriteIssuesOrPulls(comment.Issue.IsPull)) {
 		ctx.Error(http.StatusForbidden)
 		return
-	} else if comment.Type != models.CommentTypeComment && comment.Type != models.CommentTypeCode {
+	}
+	if comment.Type != models.CommentTypeComment && comment.Type != models.CommentTypeReview {
```
Comment (delvh) [inline routers/web/repo/issue.go]: What happened to the `comment.Type != models.CommentTypeCode`?
Won't that break something else?
Or was that not supposed to be present?

Comment (wxiaoguang) [reply routers/web/repo/issue.go]: Hmm, lost

Comment (wxiaoguang) [reply routers/web/repo/issue.go]: Fixed

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
trigger bot

**Pull request merged**